code-challenges.test.js

```diff
@@ -15,35 +15,169 @@
 
 // a) Create a test with an expect statement using the variable provided. HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
 
-const colors1 = ["purple", "blue", "green", "yellow", "pink"]
-// Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
-// Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+describe('shiftAndShuffle', () => {
+    const expected = ["yellow", "blue", "pink", "green"] || ["blue", "pink", "green", "yellow"] || ["pink", "green", "yellow", "blue"] || ["green", "yellow", "blue", "pink"];
+    const expected2 = ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"] || ["aqumarine", "periwinkle", "indigo", "ochre", "saffron"] || ["periwinkle", "indigo", "ochre", "saffron", "aquarmaine"] || ["indigo", "ochre", "saffron", "aquarmaine", "periwinkle"] || ["ochre", "saffron", "aquarmaine", "periwinkle","indigo"]
+    it('removes the first item from the array and shuffles the remaining content', () => {
+        const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+        // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+        const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+        // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+      expect(shiftAndShuffle (colors1)).toEqual(expect.arrayContaining(expected));
+      expect(shiftAndShuffle (colors2)).toEqual(expect.arrayContaining(expected2))
+    });
+  })
+
+
+//  ReferenceError: shiftAndShuffle is not defined
+
 
 
 // b) Create the function that makes the test pass.
 
+//music break: I had to listen to Beyonce's Break My Soul to prepare my mind 
+
+//-------Psuedo Code------//
+  // declare a function called shiftAndShuffle 
+  // set param (array)
+  // declare a variable called getShifty and assign it to the array with the first value being removed. 
+  // use the Fisher-Yates algorithm to shuffle the array.
+
+  const shiftAndShuffle = (array) => {
+    const getShifty = array.shift() 
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i+1));
+        const temp = array [i];
+        array[i] = array [j]
+        array[j] = temp;
+    }
+    return array;
+  }
+
+// Test Suites: 1 passed, 1 total
+
+
+// I did research on the Fisher-Yates algorithm and I kind of understand what is happening but could you explain it to me in laymen's terms? I would really appreciate it :)
+
```

Comment:
I feel like this is this provides the best explanation:  https://sebhastian.com/fisher-yates-shuffle-javascript/.  It has diagrams.  Plus it's a lot of words to type.  Please let me know if you have questions after reviewing this article.